### ISSUE TITLE
[Accessibility] - Group ticket

### ISSUE THREAD

**Issue by transkamp** [conterra/mapapps-query-builder#60]
the following accessibility problems exist

1. Contrast problem in autocomplete list (autumn theme)
![image](https://user-images.githubusercontent.com/15154156/212713695-00782eb0-e012-4f7c-9428-c32d374a8039.pn

2. "Delete" button is not focusable in keyboard navigation
![image](https://user-images.githubusercontent.com/15154156/212714079-5906be96-51fb-4dd5-9b16-b7fae4fdec87.png)

3. Dropdown lists are not rendered correctly in the screen reader (NVDA). The selected element is rendered as empty.
![image](https://user-images.githubusercontent.com/15154156/212714997-38dcbc41-08a9-41be-a3d5-0f0f035d001a.png)

4. label and legend elements are missing in the form fields
![image](https://user-images.githubusercontent.com/15154156/212715799-e5875895-8b4e-42f8-b40b-59bfc39753b1.png)
more infos: https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/

**Comment by sholtkamp** [comment]
3. & 4. Have been implemented into the Bundle.
1. & 2. Cannot be included as of now due to technical limitations.
